File: app/src/main/java/tpcreative/co/qrscanner/model/EnumThemeMode.java
package tpcreative.co.qrscanner.model;

public enum EnumThemeMode {
    LIGHT(0),
    DARK(1),
    DEFAULT(2);
    int positionOrdinal = 0;
    EnumThemeMode(int ord) {
        this.positionOrdinal = ord;
    }
    public static EnumThemeMode byPosition(int ord) {
        return EnumThemeMode.values()[ord]; // less safe
    }
}

File: app/src/main/java/tpcreative/co/qrscanner/model/EnumThemeMode.kt
package tpcreative.co.qrscanner.model

enum class EnumThemeMode(ord: Int) {
    LIGHT(0), DARK(1), DEFAULT(2);

    var positionOrdinal = 0

    companion object {
        fun byPosition(ord: Int): EnumThemeMode? {
            return values()[ord] // less safe
        }
    }

    init {
        positionOrdinal = ord
    }
}